### ISSUE TITLE
Behaviour change from inquirer 10.2.1 to 10.2.2

### ISSUE THREAD

**Issue by ansf** [SBoudrias/Inquirer.js#1555]
With inquirer 10.2.1 the following snippet would not prompt and just return `undefined`.
With inquirer 10.2.2 however, the prompt is shown and the value is returned accordingly.

Since this was a patch release, I think this change was not intentional.
It is breaking our scripts where we use `when` to disable prompts depending on the answer of previous prompts.

Reproducer:

```
import inquirer from 'inquirer'

const { tag } = await inquirer.prompt([
    {
        name: 'tag',
        type: 'confirm',
        message: 'Add tags?',
        default: true,
        when: () => undefined
    }
])
```

Run with inquirer 10.2.1 and 10.2.2 and observer the difference in behaviour (I can provide a full working reproducer as a repository if that helps).

**Comment by SBoudrias** [comment]
Thanks for the report, the fix is out in `inquirer@11.0.2`.

v11 isn't an API breaking change, we just reviewed some design decision in look and feel. So I'm hoping it'll be easy to update.

**Comment by ansf** [comment]
Thank you for the quick fix! I can confirm that it is now working as expected.
